hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -688,7 +688,7 @@ public void applyResultSetMapping(NamedResultSetMappingDefinition resultSetMappi
 	@Override
 	public void addDefaultResultSetMapping(NamedResultSetMappingDefinition definition) {
 		final String name = definition.getRegistrationName();
-		if ( !defaultSqlResultSetMappingNames.contains( name ) && sqlResultSetMappingMap.containsKey( name ) ) {
+		if ( !defaultSqlResultSetMappingNames.contains( name ) ) {
 			sqlResultSetMappingMap.remove( name );
```

Comment:
we don't need key checking when we simply wanna remove it, regardless of whether it exists or not.